xcm/xcm-executor/src/lib.rs

```diff
@@ -44,6 +44,12 @@ pub struct FeesMode {
 	pub jit_withdraw: bool,
 }
 
+const RECURSION_LIMIT: u8 = 10;
+
+environmental::thread_local_impl! {
+	static RECURSION_COUNT: Cell<u8> = Cell::new(0)
```

Comment:
This works in wasm?
I am literately writing some code deal with recursive calls using similar pattern right now and I am using `thread_local` for std and `static mut` for no_std. It will be cool if I can simplify the code with this.

Reply:
That's what I wanted to know as well, it seems to not be working in my case, I'll try `static mut` and see if it works...

Reply:
this is my code BTW https://github.com/open-web3-stack/open-runtime-module-library/pull/845

Reply:
Okay, I still have no idea why this isn't working. I tried changing everything into using `static mut`, but somehow I still cannot get the counter to go beyond 1, i.e. it stays at 0 at every invocation of the XCM executor. I don't believe I'm resetting it deliberately...